scripts/ocsf-schemas.py

```diff
@@ -0,0 +1,173 @@
+    "boolean_t": "bool",
+    "float_t": "double",
+    "integer_t": "int64",
+    # TODO: Don't omit this.
+    "json_t": OMIT_MARKER,
+    "long_t": "int64",
+    "string_t": "string",
+    "bytestring_t": "blob",
+    "datetime_t": "time",
+    "ip_t": "ip",
+    "subnet_t": "subnet",
+    # TODO: Is this the best choice?
+    "timestamp_t": "time",
+}
+
+# TODO: The schema defines some recursive types which cannot be represented
+# faithfully. Hence, we omit some fields to break recursive cycles. Instead of
+# that, we could recursively unfold the types up to a certain depth, or perhaps
+# use a slightly different representation.
+OMIT = {
+    "ldap_person": ["manager"],
+    "process": ["parent_process"],
+    "network_proxy": ["proxy_endpoint"],
+    "analytic": ["related_analytics"]
+}
+Schema = dict[str, dict]
+TypeMap = dict[str, str]
+
+
+def log(msg: str):
+    print("▶", msg, file=sys.stderr)
+
+
+def load_schema() -> Schema:
+    log(f"Fetching schema from {URL}")
+    return requests.get(URL).json()
+
+
+def patch_types(schema: Schema) -> None:
+    log("Patching types")
+    types = {}
+    for type_name, type_def in schema["types"].items():
+        if type_name in BASIC_TYPES:
+            result = BASIC_TYPES[type_name]
+        else:
+            result = type_def["type"]
+            if result in BASIC_TYPES:
+                result = BASIC_TYPES[result]
+        types[type_name] = result
+    schema["types"] = types
+
+
+class Writer:
+    def __init__(self, file: typing.TextIO):
+        self.file = file
+        self.indent = 0
+
+    def print(self, *args, **kwargs) -> None:
+        print(self.indent * " ", file=self.file, end="")
+        print(*args, **kwargs, file=self.file)
+
+    def comment(self, text: str) -> None:
+        width = COLUMN_LIMIT - self.indent - len("// ")
+        lines = textwrap.wrap(text, width)
+        for line in lines:
+            self.print("//", line)
+
+    def begin(self, *args, **kwargs) -> None:
+        self.print(*args, **kwargs)
+        self.indent += 2
+
+    def end(self, *args, **kwargs) -> None:
+        self.indent -= 2
+        if self.indent < 0:
+            raise ValueError
+        self.print(*args, **kwargs)
+
+
+def _emit(writer: Writer, schema: Schema, *, objects: bool) -> None:
+    name = "objects" if objects else "classes"
+    prefix = OBJECT_PREFIX if objects else CLASS_PREFIX
+    types = schema["types"]
+    first = True
+    for entity in schema[name].values():
+        if not first:
+            writer.print()
+        first = False
+        if DOCUMENT_ENTITIES:
+            writer.comment(entity["description"])
+        entity_name = entity["name"]
+        omit = OMIT.get(entity_name, [])
+        writer.begin(f"type {prefix}{entity_name} = record{{")
+        for attr_name, attr_def in sorted(entity["attributes"].items()):
+            if attr_name in omit:
+                continue
+            if "profile" in attr_def:
+                continue
```

Comment:
`"profile"`? Is this an artifact of how the schema is encoded?

Reply:
Depending on which profiles you enable, the schema changes. This makes it so that we get the same behavior as having no profiles enabled. In the future, we could provide options for this or different schemas, but I think it's a reasonable default for now.

Reply:
This is good topic. Profiles are central to OCSF. Our mappings also include them already. I see two options:
1. Start with no profiles, and accept schema fragmentation when events with new profiles arrive
2. Inject all available schemas and accept null values

If we cast our events to these schemas, I would hope that we don't lose the fields injected by the profiles.

Reply:
I made the script configurable with profiles (through code, we can later add CLI configuration but this might not be needed). The default is now to include ALL profiles. I also removed the schema definition file from the schema directory so that we do not ship this yet. We can do so at a later point in time when we actually want to use this (e.g., with the `cast` operator) and know what we want to do here.